### PR TITLE
[Storage Table]Change storage table to track2

### PR DESCRIPTION
**Pull request by Luyunmt**
Change storage table track1 to track2 and remove some unused packages.

### CODE REVIEW

**Comment by v-xuto** [comment]
@jongio - Can you help ping the corresponding person to review this PR?

**Comment by jongio** [comment]
@christothes

File: WatchdogService/CleanupOperations.cs [hunk 1]
```diff
@@ -230,22 +229,24 @@ internal async Task CleanupDiagnosticTablesAsync()
...
-                    CloudTable table = client.GetTableReference(tableName);
-                    if (true == await table.ExistsAsync(this._token))
-                    {
-                        await this.EnumerateTableItemsAsync(table);
+                    await foreach (var table in client.QueryAsync()) {
+                        if (table.Name == tableName)
+                        {
+                            var tableClient = client.GetTableClient(table.Name);
+                            await this.EnumerateTableItemsAsync(table, client);
+                        }
```
Comment (christothes) [inline WatchdogService/CleanupOperations.cs]: We could eliminate the need to query every table on each iteration.

```suggestion
                    await foreach (var table in client.QueryAsync(t => t.Name == tableName)) {
                        var tableClient = client.GetTableClient(table.Name);
                        await this.EnumerateTableItemsAsync(table, client);
```

File: WatchdogService/CleanupOperations.cs [hunk 2]
```diff
@@ -230,22 +229,24 @@ internal async Task CleanupDiagnosticTablesAsync()
...
                 {
-                    CloudTable table = client.GetTableReference(tableName);
-                    if (true == await table.ExistsAsync(this._token))
-                    {
-                        await this.EnumerateTableItemsAsync(table);
+                    await foreach (var table in client.QueryAsync()) {
+                        if (table.Name == tableName)
+                        {
+                            var tableClient = client.GetTableClient(table.Name);
+                            await this.EnumerateTableItemsAsync(table, client);
```
Comment (christothes) [inline WatchdogService/CleanupOperations.cs]: We could pass the tableClient to EnumerateTableItemsAsync
```suggestion
                            var tableClient = client.GetTableClient(table.Name);
                            await this.EnumerateTableItemsAsync(table, tableClient);
```

File: WatchdogService/CleanupOperations.cs [hunk 3]
```diff
@@ -261,7 +262,7 @@ internal async Task CleanupDiagnosticTablesAsync()
         /// </summary>
         /// <param name="table">CloudTable instance.</param>
         /// <returns>Number of items removed.</returns>
-        internal async Task<int> EnumerateTableItemsAsync(CloudTable table)
+        internal async Task<int> EnumerateTableItemsAsync(TableItem table, TableServiceClient client)
```
Comment (christothes) [inline WatchdogService/CleanupOperations.cs]: If we accept the TableClient, we have all the context needed.
```suggestion
        internal async Task<int> EnumerateTableItemsAsync(TableClient client)
```

File: WatchdogService/CleanupOperations.cs [hunk 4]
```diff
@@ -270,140 +271,23 @@ internal async Task<int> EnumerateTableItemsAsync(CloudTable table)
 
             Stopwatch sw = Stopwatch.StartNew();
             ServiceEventSource.Current.Trace("EnumerateTableItemsAsync", table.Name);
```
Comment (christothes) [inline WatchdogService/CleanupOperations.cs]:
```suggestion
            ServiceEventSource.Current.Trace("EnumerateTableItemsAsync", tableClient.Name);
```

File: WatchdogService/CleanupOperations.cs [hunk 5]
```diff
@@ -270,140 +271,23 @@ internal async Task<int> EnumerateTableItemsAsync(CloudTable table)
...
+            TableClient tableClient = client.GetTableClient(table.Name);
```
Comment (christothes) [inline WatchdogService/CleanupOperations.cs]: can be removed

File: WatchdogService/CleanupOperations.cs [hunk 6]
```diff
@@ -270,140 +271,23 @@ internal async Task<int> EnumerateTableItemsAsync(CloudTable table)
...
-
-            string pKey = null;
             int deleteCount = 0;
-            TableContinuationToken tct = null;
-            TableQuery query =
-                new TableQuery().Where(
-                    TableQuery.GenerateFilterConditionForDate("Timestamp", QueryComparisons.LessThan, DateTimeOffset.Now.Subtract(this._timeToKeep)));
+            var timeStamp = DateTimeOffset.Now.Subtract(this._timeToKeep);
+            // Execute the query.
+            AsyncPageable<TableEntity> queryResult = tableClient.QueryAsync<TableEntity>(filter: $"Timestamp lt '{timeStamp}'");
```
Comment (christothes) [inline WatchdogService/CleanupOperations.cs]: We could use filter expression syntax here
```suggestion
            AsyncPageable<TableEntity> queryResult = tableClient.QueryAsync<TableEntity>(e => e.Timestamp < timeStamp);
```

File: WatchdogService/packages.config [hunk 7]
```diff
@@ -1,5 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
+  <package id="Azure.Core" version="1.20.0" targetFramework="net461" />
+  <package id="Azure.Data.Tables" version="12.2.0" targetFramework="net461" />
```
Comment (christothes) [inline WatchdogService/packages.config]:
```suggestion
  <package id="Azure.Data.Tables" version="12.2.1" targetFramework="net461" />
```

**Review by christothes — COMMENTED** [review]
Left a few suggestions

File: WatchdogService/CleanupOperations.cs (hunk 5 above)
Comment (Tianyun-G) [reply WatchdogService/CleanupOperations.cs]: @christothes - I have updated, please review.

File: WatchdogService/WatchdogService.csproj [hunk 8]
```diff
@@ -38,6 +39,12 @@
     <AdditionalFileItemNames>$(AdditionalFileItemNames);None</AdditionalFileItemNames>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="Azure.Core, Version=1.20.0.0, Culture=neutral, PublicKeyToken=92742159e12e44c8, processorArchitecture=MSIL">
+      <HintPath>..\packages\Azure.Core.1.20.0\lib\net461\Azure.Core.dll</HintPath>
+    </Reference>
+    <Reference Include="Azure.Data.Tables, Version=12.2.0.0, Culture=neutral, PublicKeyToken=92742159e12e44c8, processorArchitecture=MSIL">
+      <HintPath>..\packages\Azure.Data.Tables.12.2.0\lib\netstandard2.0\Azure.Data.Tables.dll</HintPath>
```
Comment (christothes) [inline WatchdogService/WatchdogService.csproj]: The assembly reference doesn't appear to have been updated to reflect the package version

Comment (Tianyun-G) [reply WatchdogService/WatchdogService.csproj]: @christothes - I have updated, please review.

**Review by christothes — APPROVED** [review]

**Pull request merged**